src/d3a/models/market/blockchain_interface.py

```diff
@@ -59,11 +65,88 @@ def bc_listener(self):
         pass
 
 
+class SubstrateBlockchainInterface():
+    def __init__(self):
+        self.contracts = {'main': main_address}
+        self.substrate = SubstrateInterface(
+            url="ws://127.0.0.1:9944",
+            address_type=42,
+            type_registry_preset='default'
+        )
+
+    def simple_contract_call(self, contract_address, function, path):
+        data = get_function_metadata(function, path)
+        call = self.substrate.compose_call(
+            call_module='Contracts',
+            call_function='call',
+            call_params={
+                'dest': contract_address,
+                'value': 1 * 10**18,
+                'gas_limit': 1000000000000,
+                'data': data
+            }
+        )
+        return call
+
+    def storage_contract_call(self, contract_address, function, path):
+        trade_id = random.randint(10, 16)
+        print('TRADE ID {}'.format(trade_id))
```

Comment:
Prints are indeed handy while developing code but please replace it with log messages before merging this PR. Thanks!

Reply:
Changed to logging